### PR TITLE
Add itoa implementation

### PR DESCRIPTION
**Pull request by fjeremic**

### CODE REVIEW

File: itoa.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+// Converts an integer to a dynamically allocated null-terminated C string
+// Examples:
+//            15 -> ['1', '5', '\0']
+//            -5 -> ['-', '5', '\0']
+//           123 -> ['1', '2', '3', '\0']
+int getNumDigits(int v) {
+    int numDigits = 0;
+    do {
+        numDigits++;
+        v = v / 10;
```
Comment (singh264) [inline itoa.c]: Divide operation operation can be replaced using shift operation for better performance

File: itoa.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            lsd *= -1;
+        }
+
+        char lsdc = (char) ('0' + lsd);
+
+        vstr[vstrIndex] = lsdc;
+
+        vstrIndex--;
+
+        vtemp = vtemp / 10;
```
Comment (singh264) [inline itoa.c]: Divide operation operation can be replaced using shift operation for better performance

File: itoa.c [hunk 3]
```diff
@@ -0,0 +1,81 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+// Converts an integer to a dynamically allocated null-terminated C string
+// Examples:
+//            15 -> ['1', '5', '\0']
+//            -5 -> ['-', '5', '\0']
+//           123 -> ['1', '2', '3', '\0']
+int getNumDigits(int v) {
```
Comment (singh264) [inline itoa.c]: To be consistent with `itoa(..)`
```suggestion
int getNumDigits(int v) 
{
```

File: itoa.c (hunk 1 above)
Comment (fjeremic) [reply itoa.c]: How does one divide by 10 using shifts? Do you think the compiler already optimizes such divisions on our behalf?

Comment (singh264) [reply itoa.c]: We cannot divide-by-10 with just shift operations. We can replace the divide-by-10 with one multiply operation and one shift operation. [1] I noticed a 1.17x (= 3.87sec / 3.31sec) speedup with the modified divby10 below. 

```
int divby10(int v) {
    long intDivisor = 0x1999999A;
    int res = (intDivisor * v) >> 32;
    if (v < 0) {
        res++;
    }
    return res;
}
```

Unless all compiler optimizations are disable, the compiler would optimize such division on our behalf. We can verify this by inspecting the assembly code in the objdump file.

[1] [A close approximation of 1/10 * 2^32 and remove 2^32 with a shift operation](https://stackoverflow.com/questions/5558492/divide-by-10-using-bit-shifts/34275429).

Comment (fjeremic) [reply itoa.c]: Excellent analysis. Thanks!

**Pull request closed without merging**